coremltools/converters/mil/frontend/torch/test/test_torch_ops.py

```diff
@@ -2824,17 +2866,22 @@ def forward(self, x, y):
             input_shapes = [[1,2,3,4], [1,4,2,6]]
         elif equation == "abcd,cde->abe":
             input_shapes = [[1,2,3,4], [3,4,6]]
+
+        # Generic cases
         else:
-            raise ValueError("unrecognized equation")
+            input_shapes, converter_input_type = make_input_types(equation)
```

Comment:
This is a great utility function, thanks for adding it.
I think it would be better for "both" hardcode and genric cases,
we can test both static and dynamic shape,
now we are testing static shape for hardcoded case and dynamic shape for generic case.

I think a better way to do it is to put the `dynamic = [True, False]` in the pytest's `parametrize`,
so we can get rid of line 2853 -> 2868,
and using your `make_input_types`function.

You only need to add a new boolean argument `dynamic` in the `make_input_types` function to produce both dynamic and static input shape.

Reply:
Thx! I added test for both static and dynamic cases for all equation.
And I found some hardcode branches has not supported flexible shape.
They use `a_var.shape[0]`.

I changed such branches to use `solve_generic_einsum` if flexible shape (See _utils.py l.86 and l.107).

Reply:
Great change